src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,198 @@
+
+     /// @notice variable that defines the share EigenLayer takes of all payments, in basis points
+     uint256 public eigenLayerShareBIPs;
+
+    // TODO: better define this event?
+    event PaymentReceived(address indexed receivedFrom, Payment payment);
+
+    // @notice Emitted when a new Merkle root is posted
+    event NewMerkleRootPosted(MerkleRootPost merkleRootPost);
+
+    event PaymentClaimed(MerkleLeaf merkleLeaf);
+
+    /// @notice Emitted when the rootPublisher is changed
+    event RootPublisherChanged(address indexed oldRootPublisher, address indexed newRootPublisher);
+
+    /// @notice Emitted when the merkle root activiation delay is changed
+    event MerkleRootActivationDelayBlocksChanged(uint256 newMerkleRootActivationDelayBlocks);
+
+
+    /// @notice Emitted when the EigenLayer's percentage share is changed
+    event EigenLayerShareBIPSChanged(uint256 newEigenLayerShareBIPS);
+
+    modifier onlyRootPublisher {
+        require(msg.sender == rootPublisher, "PaymentCoordinator: Only rootPublisher");
+        _;
+    }
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(address _initialOwner, address _rootPublisher, uint256 _eigenlayerShareBips, uint256 _merkleRootActivationDelayBlocks)
+        external
+        initializer
+    {
+        _transferOwnership(_initialOwner);
+        _setRootPublisher(_rootPublisher);
+        _setMerkleRootActivationDelayBlocks(_merkleRootActivationDelayBlocks);
+        _setEigenLayerShareBIPS(_eigenlayerShareBips);
+    }
+
+
+    /**
+     * @notice Makes a payment of sum(amounts) paid in `token`, for `operator`'s contributions to an AVS,
+     * between `startBlockNumber` (inclusive) and `endBlockNumber` (inclusive)
+     * @dev Transfers the total payment from the `msg.sender` to this contract, so the caller must have previously approved
+     * this contract to transfer at least sum(`amounts`) of `token`
+     * @notice Emits a `PaymentReceived` event
+     */
+    function makePayment(Payment calldata payment) external{
+        require(payment.amounts.length == payment.quorums.length, "PaymentCoordinator.makePayment: payment amounts and quorums must have the same length");
+        uint256 sumAmounts;
+        for (uint256 i = 0; i < payment.amounts.length; i++) {
+            sumAmounts += payment.amounts[i];
+        }
+        payment.token.safeTransferFrom(msg.sender, address(this), sumAmounts);
+
+        accumulatedEigenLayerTokenEarnings[payment.token] += sumAmounts * eigenLayerShareBIPs / MAX_BIPS;
+        emit PaymentReceived(msg.sender, payment);
+    }
+
+    // @notice Permissioned function which allows posting a new Merkle root
+    function postMerkleRoot(bytes32 newRoot, uint256 height, uint256 calculatedUpToBlockNumber) external onlyRootPublisher {
+        MerkleRootPost memory newMerkleRoot = MerkleRootPost(newRoot, height, block.number + merkleRootActivationDelayBlocks, calculatedUpToBlockNumber);
```

Comment:
I'd prefer the declaration explicitly list the struct fields, like:
```
MerkleRootPost memory newMerkleRoot = MerkleRootPost(
{
root: newRoot
...
}
);
```
I think it is clearer / easier to audit / harder to mess up.